File: content/examples/other/leetcode/problems/unique-paths/example-2/Solution.java
import java.util.Arrays;

public class Solution {
  static int uniquePaths1D(int m, int n) {
    int[] dp = new int[n];
    Arrays.fill(dp, 1);
    for (int r = 1; r < m; r += 1) {
      for (int c = 1; c < n; c += 1) dp[c] = dp[c] + dp[c - 1];
    }
    return dp[n - 1];
  }

  public static void main(String[] args) {
    System.out.println(uniquePaths1D(3, 7));
  }
}

File: content/examples/other/leetcode/problems/unique-paths/example-2/solution.js
function uniquePaths1D(m, n) {
  const dp = new Array(n).fill(1);
  for (let r = 1; r < m; r += 1) {
    for (let c = 1; c < n; c += 1) dp[c] = dp[c] + dp[c - 1];
  }
  return dp[n - 1];
}

if (require.main === module) {
  console.log(uniquePaths1D(3, 7));
}

module.exports = { uniquePaths1D };
